test/with-typescript/index.ts

```diff
@@ -8,10 +8,16 @@ import NextJS from "../../nextjs";
 import { RecipeImplementation as FaunaDBImplementation } from "../../recipe/session/faunadb";
 let faunadb = require("faunadb");
 import ThirdPartyEmailPassword from "../../recipe/thirdpartyemailpassword";
+import ThirdParty from "../../recipe/thirdparty";
 import Passwordless from "../../recipe/passwordless";
 import ThirdPartyPasswordless from "../../recipe/thirdpartypasswordless";
-import { STMPService } from "../../recipe/thirdpartypasswordless/emaildelivery";
-import { TwilioService } from "../../recipe/thirdpartypasswordless/smsdelivery";
+import { STMPService as STMPServiceTPP } from "../../recipe/thirdpartypasswordless/emaildelivery";
+import { STMPService as STMPServiceP } from "../../recipe/passwordless/emaildelivery";
+import { STMPService as STMPServiceTP } from "../../recipe/thirdparty/emaildelivery";
+import { STMPService as STMPServiceTPEP } from "../../recipe/thirdpartyemailpassword/emaildelivery";
+import { STMPService as STMPServiceEP } from "../../recipe/emailpassword/emaildelivery";
+import { TwilioService as TwilioServiceTPP } from "../../recipe/thirdpartypasswordless/smsdelivery";
+import { TwilioService as TwilioServiceP } from "../../recipe/thirdpartypasswordless/smsdelivery";
 
```

Comment:
- Missing for where you give service AND overrride together..
- Missing checking the types inside the `input` variable when providing the override function everywhere.